python/foglamp/common/process.py

```diff
@@ -29,6 +29,13 @@ class ArgumentParserError(Exception):
     """ Overwrite default exception to not terminate application """
     pass
 
+class SilentArgParse(argparse.ArgumentParser):
+    def error(self, message):
+        raise ArgumentParserError(message)
+    def silent_arg_parse(self, argument_name):
+        self.add_argument(argument_name)
+        parser_result = self.parse_known_args()
```

Comment:
should not we move this to try except? how error def is being used. 

should not we just remove def error! and:

```  try:
          parser_result = parser.parse_known_args()
       except ArgumentParserError:
     raise
       else:
          return list(vars(parser_result[0]).values())[0]
```

Reply:
The regular error method from argparse exits the application if there is an error. We simply want to raise an exception, so I have overwritten the method.

Reply:
I see, this is overridden method. Please add a comment. thanks for clarification.